aten/src/ATen/native/cuda/TensorFactories.cu

```diff
@@ -26,4 +36,57 @@ Tensor& eye_out_cuda(Tensor& result, int64_t n, int64_t m) {
   return result;
 }
 
+Tensor& randperm_out_cuda(Tensor& result, int64_t n, Generator* generator) {
+  if (n < 0) {
```

Comment:
Why do you use `AT_CHECK` below, but not here? We should try to use `AT_*` macros everywhere, as they include stack traces. Could have been
```
AT_CHECK(n >= 0, "n must be non-negative, got ", n);
```
```